### PR TITLE
feat: integrate flood control addon into core

### PR DESCRIPTION
**Pull request by mussbach**
This PR integrates the FloodControl Addon into core repository.

The reasoning behind this is that the addon was enabled by default in every project and therefore obviously a core feature. Moreover, as it belongs to the features we need to publish the addon needs to be set under the EuPL as well. We could decide later on to remove the FloodControl from core again once the addon system is mature enough, but for now there is no advantage to keep it as a separate addon, but instead it creates problems at the moment we could avoid.

ToDo
- [x] Test when SegmentsManager addon is back to core, so that FloodControl could be reached

### How to review/test
Check Floodcontrol functionality like login too rapidly

### PR Checklist

Delete the checkbox if it doesn't apply/isn't necessary.

- [x] Tests updated/created
- [ ] Update documentation
- [ ] Link all relevant tickets
- [ ] Move the tickets on the board accordingly

### CODE REVIEW

File: demosplan/DemosPlanCoreBundle/Logic/FloodControlService.php [hunk 1]
```diff
@@ -0,0 +1,271 @@
...
+
+        if ($request->request->has('url') && mb_strlen(trim($request->request->get('url'))) > 0) {
+            $this->getLogger()->info('FloodControl: Honeypot field has been filled out');
+            throw new HoneypotException('Honeypot field has been filled out');
+        }
+
+        if ($request->request->has('r_loadtime')) {
+            $loadtime = $request->request->get('r_loadtime');
+            $totaltime = time() - $loadtime;
+            if ($totaltime < $this->globalConfig->getHoneypotTimeout()) {
```
[the remark this answers is not in the document]
Comment (mussbach) [reply demosplan/DemosPlanCoreBundle/Logic/FloodControlService.php:104]: Needs to be integrated in Contract, this is done in another PR in the contracts repository. Method is included in `GobalConfig`

File: demosplan/DemosPlanCoreBundle/Controller/Statement/DemosPlanStatementController.php [hunk 2]
```diff
@@ -70,23 +67,8 @@
...
 use demosplan\DemosPlanUserBundle\Logic\OrgaHandler;
 use demosplan\DemosPlanUserBundle\Logic\UserService;
-
-use const ENT_QUOTES;
-
 use Exception;
-
-use function explode;
-use function html_entity_decode;
-use function implode;
```
Comment (breierDemos) [inline demosplan/DemosPlanCoreBundle/Controller/Statement/DemosPlanStatementController.php:80]: Why was the use statements of the functions removed? In other files they still remaining. The calls to the functions are supplemented by a "\\". According to what rule do we use it like this?

**Review by breierDemos — CHANGES_REQUESTED** [review]
Lgtm so far, Tests run successfully.
One note and one question.

File: demosplan/DemosPlanCoreBundle/Logic/FloodControlService.php [hunk 3]
```diff
@@ -0,0 +1,270 @@
...
+                $this->getLogger()->info('FloodControl: Honeypot form has been submitted too fast');
+                $this->messageBag->add(
+                    'warning',
```
[the remark this answers is not in the document]
Comment (mussbach) [reply demosplan/DemosPlanCoreBundle/Logic/FloodControlService.php]: Indeed, this is possible now. Thank you 930a4b41

**Review by breierDemos — APPROVED** [review]

**Pull request merged**